image/helpers.go

```diff
@@ -342,13 +347,96 @@ func (tsto *ToolingStore) DownloadSnap(name string, opts DownloadOptions) (targe
 	}()
 
 	dlOpts := &store.DownloadOptions{LeavePartialOnError: opts.LeavePartialOnError}
-	if err = sto.Download(context.TODO(), name, targetFn, &snap.DownloadInfo, pb, tsto.user, dlOpts); err != nil {
-		return "", nil, "", err
+	if err = tsto.sto.Download(context.TODO(), snap.SnapName(), targetFn, &snap.DownloadInfo, pb, tsto.user, dlOpts); err != nil {
+		return nil, err
 	}
 
 	signal.Reset(syscall.SIGINT)
 
-	return targetFn, snap, redirectChannel, nil
+	return &DownloadedSnap{
+		Path:            targetFn,
+		Info:            snap,
+		RedirectChannel: redirectChannel,
+	}, nil
+}
+
+type SnapDownloadOptions struct {
+	Snap      naming.SnapRef
+	Channel   string
+	CohortKey string
+}
+
+type CurrentSnap struct {
+	SnapName string
+	SnapID   string
+	Revision snap.Revision
+	Channel  string
+	Epoch    snap.Epoch
+}
+
+type DownloadManyOptions struct {
+	BeforeDownloadFunc func(*snap.Info) (targetPath string, err error)
+	EnforceValidation  bool
+}
+
+// DownloadMany downloads the specified snaps.
```

Comment:
Could you add a comment explaining what `curSnaps` is? I understant that `toDownload` is clearly the list of snaps to be downloaded, but `curSnaps` is more subtle.